### PR TITLE
Support to list and get route53 resource records from a hosted zone

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: route53/records_test.go [hunk 1]
```diff
@@ -176,3 +267,135 @@ func TestCreateRecord(t *testing.T) {
...
+		Service: newmockRoute53Client(t, nil),
+		Domains: map[string]common.Domain{
+			"hyper.converged": common.Domain{
+				CertArn: "arn:aws:acm::12345678910:certificate/111111111-2222-3333-4444-555555555555",
+			},
+		},
+	}
+
+	expected := &testResourceRecordSet
+	out, err := r.GetRecord(context.TODO(), testHostedZoneID, "foobar.hyper.converged", "A")
```
Comment (tenyo) [inline route53/records_test.go]: Is it worth testing when the host name ends in a `.` it should also succeed?

File: route53/records_test.go [hunk 2]
```diff
@@ -58,6 +89,66 @@ func (m *mockRoute53Client) ChangeResourceRecordSetsWithContext(ctx context.Cont
 	}, nil
 }
 
+func (m *mockRoute53Client) ListResourceRecordSetsWithContext(ctx context.Context, input *route53.ListResourceRecordSetsInput, opts ...request.Option) (*route53.ListResourceRecordSetsOutput, error) {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records_test.go]: yeah man #mockLikeaBoss

File: route53/records.go [hunk 3]
```diff
@@ -33,3 +37,56 @@ func (r *Route53) CreateRecord(ctx context.Context, zoneID string, record *route
...
+	log.Infof("getting route53 record for zone ID %s, host %s", zoneID, host)
+
+	records, err := r.ListRecords(ctx, zoneID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range records {
+		if !strings.HasSuffix(host, ".") {
+			host = host + "."
```
[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records.go]: good catch

File: route53/records.go [hunk 4]
```diff
@@ -33,3 +37,56 @@ func (r *Route53) CreateRecord(ctx context.Context, zoneID string, record *route
...
+// GetRecord gets a route53 resource record
+func (r *Route53) GetRecord(ctx context.Context, zoneID, host, recordType string) (*route53.ResourceRecordSet, error) {
+	log.Infof("getting route53 record for zone ID %s, host %s", zoneID, host)
+
+	records, err := r.ListRecords(ctx, zoneID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range records {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records.go]: yeah good catch

File: route53/records_test.go [hunk 5]
```diff
@@ -176,3 +267,135 @@ func TestCreateRecord(t *testing.T) {
...
+		t.Errorf("expected nil error, got: %s", err)
+	}
+
+	if !reflect.DeepEqual(out, expected) {
+		t.Errorf("expected %+v, got %+v", expected, out)
+	}
+
+	// test with wrong name
+	out, err = r.GetRecord(context.TODO(), testHostedZoneID, "foobaz.hyper.converged", "A")
+	if err == nil || out != nil {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records_test.go]: i want it to fail the test if there is no error or if out is anything but nil

File: route53/records_test.go (hunk 1 above)
Comment (fishnix) [reply route53/records_test.go]: probably, ill add it

File: route53/records_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (tenyo) [reply route53/records_test.go]: yeah, but don't you use `err` in the next block?

[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records_test.go]: only if its not nil

[the remark this answers is not in the document]
Comment (tenyo) [reply route53/records_test.go]: i mean, if `err != nil` and `out != nil` this test will fail - even though it returns an error

[the remark this answers is not in the document]
Comment (fishnix) [reply route53/records_test.go]: that's what I want, out should be nil

[the remark this answers is not in the document]
Comment (tenyo) [reply route53/records_test.go]: 👍

**Comment by damnski** [comment]
:+1:

**Comment by tenyo** [comment]
👍

**Pull request merged**